### ISSUE TITLE
There should be a space between the letter and the number on Dutch highway shields

### ISSUE THREAD

**Issue by louwers** [osm-americana/openstreetmap-americana#1148]
![image](https://github.com/user-attachments/assets/61f49128-2d2f-4352-9eb8-400bb0c5184b)

**Comment by claysmalley** [comment]
The Americana renderer takes no stance on matters of local formatting and punctuation—the `ref=*` value is displayed verbatim, as it is tagged in the OSM database. So the Dutch OSM community would need to agree to retagging all the relations.

cc @Blijbol

**Comment by louwers** [comment]
@claysmalley Well, the official names are without a space, and when referring to highways in text we also don't use a space. So there is no chance this would garner any support from the Dutch OSM community.

However if your cartographic goal is to have accurate representations of highway shields (which I think is the case), you should use a space.

**Comment by claysmalley** [comment]
> However if your cartographic goal is to have accurate representations of highway shields (which I think is the case), you should use a space.

Our goal with shields is not necessarily to have a perfect copy of what's on the ground, but to  stylize and simplify a little so that they look good on a digital map, and provide a data-driven renderer that displays shields for `route=road` relations, however the local community has tagged them. It's up to the local community to adopt a tagging scheme that matches shields, or not.

Near where I live, there is a highway called Interstate 540, or I-540 for short. The [relation for I-540](https://www.openstreetmap.org/relation/376238) is tagged `ref=540`, because that is the text present on signage, and we've chosen as a community to match `ref=*` to the text on signs. The "official name" isn't in the relation at all.

**Comment by ZeLonewolf** [comment]
Based on the initial photograph of the overhead signage (not the vector graphic below it), I am not seeing a space between the letter and number, or if there is a space, it is very very tiny. If you compare the A50 in the photograph to the A59 in the vector graphic, the space between "A" and "59" is much larger than the space between the "A" and the "50". Just eyeballing things, I think the vector graphic is putting a bigger space in than the signs say in reality.

All this to say -- I am not convinced there should even be a space between the letter and number based on what I'm seeing here.

**Comment by louwers** [comment]
![972868](https://github.com/user-attachments/assets/688ca7a9-0854-4647-9928-7be34ad83ef1)

![image](https://github.com/user-attachments/assets/b9889cbc-bc25-456a-bedd-f766366c692c)

@ZeLonewolf Right. I don't think it is a space. I sent an inquiry to [Rijkswaterstaat](https://en.wikipedia.org/wiki/Rijkswaterstaat) and asked for clarification.

**Comment by ZeLonewolf** [comment]
I would also point out our style [contributor's guide](https://github.com/osm-americana/openstreetmap-americana/blob/main/CONTRIBUTING.md), which has this to say about shield design:

> In general, this style is not trying to exactly replicate highway shields as seen on signage. Instead, we are trying to extract the key stylistic elements so that the graphics are recognizable as simplifications of their real-world counterparts.

Some number of years ago, @ke9tv developed a map with photo-realistic highway shields with extraordinary attention to detail. Unfortunately, images of this map seem to be lost to time.

His map even went to the level of detail of rendering shields where suffix letters were drawn in a smaller font than the main route number, as shown in the "5A" and "5B" examples here:

![image](https://github.com/user-attachments/assets/53eeb14a-cc8a-4b6a-8a9e-2990333f5efc)

We have not, so far in this project, attempted to micro-style these differences, as shown in our on-map rendering of New York state route 5A:

![image](https://github.com/user-attachments/assets/392c746c-8843-4798-9b52-02bfdc97bf7b)

The only exception I can think of where we have consensus for micro-styling of text in this project is #366, which proposes to handle cases where shields have two rows of text by rendering the `/` character which appears in the `ref` tag as a line break. However, in those cases, the contents of the `ref` tag alone provide the information needed for the renderer to produce the line break behavior in a general way.

I would consider this describe issue to be closer in nature to the New York case, which is one example of a class of text renderings that have minor stylistic adjustments on the real-world signs that we wouldn't produce a distinction for on the real-world map, if indeed there is a "less than a space" gap between characters.

There are other route networks where the space is explicit, both on signage and in the map:

![image](https://github.com/user-attachments/assets/cebc9b37-a968-457e-be51-ca1dd5a79c40)

If we find that in the Dutch case, spaces are being introduced in signage, this suggests that spaces should be inserted in the tagging as well. In the more recent photographs, it appears that the Dutch routes are following the text layout of European E-routes, which do place the space in the `ref` tag.

In summary, I agree with @claysmalley's assessment that this is a tagging issue to be sorted out with the Dutch mapping community rather than a styling requirement that the renderer should address.

**Comment by Blijbol** [comment]
According to the [_Nationale routenummering_ (Rijkswaterstaat, 1982)](https://open.rijkswaterstaat.nl/zoeken/@157458/nationale-routenummering/), the correct way to write the numbers is without space (page 5):

> Het verdient dan ook aanbeveling te spreken of te schrijven over bijvoorbeeld de A12, de N264, de s105 en eventueel over de E35.

The `ref`s in the Netherlands are tagged as such (except s-roads nowadays use uppercase S, which was an actual change at some point in time).

However, for readability there has always been slight spacing on road signs, as you can see in figures 2.1 through 2.4 of the same document. The exact amount of spacing varies depending on when the road sign was designed and by whom (there have long been [major differences between Rijkswaterstaat and ANWB](https://www.wegenwiki.nl/Redesign)). However, the spacing is always less than a full character and can't be considered a proper space character. Meanwhile [milestone signs show no spacing at all](https://www.google.com/maps/@51.9542901,4.5566822,3a,15y,108.72h,78.27t/data=!3m7!1e1!3m5!1sNBLpcjtpLHHJvLDbhyOKKg!2e0!6shttps:%2F%2Fstreetviewpixels-pa.googleapis.com%2Fv1%2Fthumbnail%3Fcb_client%3Dmaps_sv.tactile%26w%3D900%26h%3D600%26pitch%3D11.729755577457283%26panoid%3DNBLpcjtpLHHJvLDbhyOKKg%26yaw%3D108.71882465516327!7i16384!8i8192?coh=205410&entry=ttu).

So this is basically a stylistic choice for the renderer, not a tagging problem. Even though Americana has no policy of exactly replicating road signs, it might be interesting for Americana to consider modifying spacing for better international consistency, e.g. by inserting/modifying spaces in `ref` values via `network`-dependent regular expressions.

**Comment by louwers** [comment]
From my subjective experience, having been exposed to these shields all my life, I cannot say that I find the Dutch highway shields in Americana particularly recognizable. 

> If we find that in the Dutch case, spaces are being introduced in signage, this suggests that spaces should be inserted in the tagging as well.

Disagreed. The [official references](https://www.rijkswaterstaat.nl/wegen/wegenoverzicht) of these highways are without a space. Adding a space would be a case of Tagging for the Renderer.

**Comment by ZeLonewolf** [comment]
> From my subjective experience, having been exposed to these shields all my life, I cannot say that I find the Dutch highway shields in Americana particularly recognizable.

These are not recognizable?
![image](https://github.com/user-attachments/assets/9cb3bf20-37a7-49ab-a7ff-f5588dc1963f)![image](https://github.com/user-attachments/assets/7623f448-83a1-4007-96a6-eea12ec3ee0e)

**Comment by 1ec5** [comment]
This came up previously in https://github.com/osm-americana/openstreetmap-americana/issues/141#issuecomment-1122797418.

In general, this style’s artistic direction pays homage to a particular tradition from print cartography. One element of that is to apply a consistent typeface and color palette to every shield from every country, even if that sometimes requires a slight reinterpretation of the design. So it’s a question of whether we can “get away with” a departure of this magnitude.

There seems to be some precedent for more or less the current treatment, which is not to say that we must uncritically follow this particular example:

![An inset map of Amsterdam with examples of A, E, and N roads as red, green, and yellow rectangles, respectively. The shield text is set in the medium weight of a geometric typeface resembling Futura, and there is no space between the alphabetic prefix and the digits.](https://github.com/user-attachments/assets/66e1f9cd-55a7-4945-8ef9-605017a6025e)

_France & Benelux Countries_ (American ed.). 1:500,000. Cartography by Freytag & Berndt. American Automobile Association. 2009. Amsterdam inset.

This is far from the only situation where we’ve taken liberties with route shields. For one thing, we made a decision early on _not_ to hew too closely to reassurance markers for state and provincial routes in North America, because they tended to be too busy for the purpose of marking a route on an already busy map. Rightly or wrongly, this informed many of the initial simplifications we made as we expanded coverage around the world.

Sign | Marker | Description
:--:|:--:|:--:
[<img src="https://upload.wikimedia.org/wikipedia/commons/7/75/Quebec_Autoroute_5.svg" height="20" alt="A-5">](https://en.wikipedia.org/wiki/File:Quebec_Autoroute_5.svg) | <img src="https://github.com/user-attachments/assets/84f5f56b-23f2-46b3-b687-e9da8c5a8458" height="20" alt="A-5"> | Québec Autoroute 5
[<img height="25" alt="249" src="https://github.com/user-attachments/assets/dfe1ab28-77d8-401e-b525-3115d91274f9">](https://commons.wikimedia.org/wiki/File:Westbound_Zion_Road_at_Texas_State_Highway_249.jpg) | <img src="https://github.com/user-attachments/assets/d79545eb-60de-47ee-b6ab-c2946a2aab79" height="20" alt="249"> | MCTRA 249 Tollway
[<img src="https://upload.wikimedia.org/wikipedia/commons/9/97/China_Expwy_G1512_sign_with_name.svg" height="20" alt="国家高速 G1512 甬金高速">](https://commons.wikimedia.org/wiki/File:China_Expwy_G1512_sign_with_name.svg) | <img src="https://github.com/user-attachments/assets/fabc1b7b-e2a8-458e-8098-fc7fca37109f" height="20" alt="G1512"> | G15<sub>12</sub> Ningbo–Jinhua Expressway

From a technical perspective, we want to gradually reduce the amount of custom string processing that the shield library does, not increase it, to improve the style’s portability in different environments, such as mobile platforms. One possible approach would be to port the custom JavaScript to MapLibre style expressions, but unfortunately the expression language lacks anything in the way of string processing operators. Don’t get me started on what it took just to be able to find and replace semicolons in #666 #670, let alone something that would seemingly require a regular expression. If anyone here is able to push through stuff like maplibre/maplibre-gl-js#2064 maplibre/maplibre-gl-js#2059, it would give us a lot more confidence about that direction. Instead, we’ve taken a different route, trying to make the shields more data-driven, for example working with OpenMapTiles to expose `ref:colour=*` tags so that we no longer need to hard-code a lookup table of colors for certain color-coded route networks: https://github.com/ZeLonewolf/openstreetmap-americana/issues/654#issuecomment-2008323861.

To the extent that these departures from the physical signs present difficulties in comprehension, hopefully the legend helps a bit as a workaround for now. And of course, just because OSM Americana prioritizes typographic consistency over pixel-perfect shield rendering, that doesn’t mean that it’s the One True Way™. There are several active forks of Americana that apply their own artistic direction. Try it out and maybe we’ll be really jealous of what you come up with and want to copy you. 😉

**Comment by 1ec5** [comment]
> This way you can also work with special Unicode space characters of various widths.

> One possible approach would be to port the custom JavaScript to MapLibre style expressions, but unfortunately the expression language lacks anything in the way of string processing operators.

On top of that, the expression operators are incorrectly implemented so that they’ll misbehave if we pass a non-ASCII space into them: maplibre/maplibre-style-spec#778 maplibre/maplibre-native#2730.

**Comment by timboehde** [comment]
For what it's worth, I am responsible for Template:Jct originally on Wikipedia and now on AARoads Wiki using a half-space between the letter and number.  It was 100% stylistic and is by no means impossible to change, it's just a default.
